aptos-move/framework/src/natives/aggregator_natives/aggregator_v2.rs

```diff
@@ -271,33 +208,32 @@ fn native_try_add(
     debug_assert_eq!(ty_args.len(), 1);
     context.charge(AGGREGATOR_V2_TRY_ADD_BASE)?;
 
-    let input = pop_value_by_type(&ty_args[0], &mut args, EUNSUPPORTED_AGGREGATOR_TYPE)?;
-    let agg_struct = safely_pop_arg!(args, StructRef);
-    let (agg_value, agg_max_value) = get_aggregator_fields_by_type(&ty_args[0], &agg_struct)?;
+    let aggregator_value_ty = &ty_args[0];
+    let rhs = pop_value_by_type(aggregator_value_ty, &mut args, EUNSUPPORTED_AGGREGATOR_TYPE)?;
+    let aggregator = safely_pop_arg!(args, StructRef);
 
-    let result_value = if let Some((resolver, mut delayed_field_data)) = get_context_data(context) {
-        let id = aggregator_value_field_as_id(agg_value, resolver)?;
-        delayed_field_data.try_add_delta(
-            id,
-            agg_max_value,
-            SignedU128::Positive(input),
-            resolver,
-        )?
+    let max_value = get_aggregator_max_value(&aggregator, aggregator_value_ty)?;
+
+    let success = if let Some((resolver, mut delayed_field_data)) = get_context_data(context) {
+        let id = get_aggregator_value_as_id(&aggregator, aggregator_value_ty, resolver)?;
+        delayed_field_data.try_add_delta(id, max_value, SignedU128::Positive(rhs), resolver)?
     } else {
-        let math = BoundedMath::new(agg_max_value);
-        match math.unsigned_add(agg_value, input) {
-            Ok(sum) => {
-                set_aggregator_value_field(
-                    &agg_struct,
-                    create_value_by_type(&ty_args[0], sum, EUNSUPPORTED_AGGREGATOR_TYPE)?,
+        let lhs = get_aggregator_value(&aggregator, aggregator_value_ty)?;
```

Comment:
I think I'm confusing myself here, but would the v2 API be enabled in that case. what does it mean to use try_add if resolver is not capable, for example

Reply:
we have two flags:
AGGREGATOR_V2_API
AGGREGATOR_V2_DELAYED_FIELDS

when first one is not enabled, we fail at the beginning of each function

if AGGREGATOR_V2_DELAYED_FIELDS is not enabled or we are not in capable context (we are not executing onchain transactions, i.e. tests), we don't exchange fields and use a regular move implementation of aggregators (fallback) - that is semantically equivalent - except it is sequential. 

our e2e test check that transaction output with and without AGGREGATOR_V2_DELAYED_FIELDS/RESOURCE_GROUPS_CHARGE_AS_SIZE_SUM flags enabled is identical

Reply:
Thanks. That's why I was asking whether it's only used in the test context, we can assert it cfg(test)

Reply:
but maybe that's not applicable to those tests, among other things.

Reply:
it is used in prod! 

we are going to first enable AGGREGATOR_V2_API, and then enable AGGREGATOR_V2_DELAYED_FIELDS. and if there is issue with delayed fields, we can always change AGGREGATOR_V2_DELAYED_FIELDS back to false with only loosing efficiency, but we can never change AGGREGATOR_V2_API to false - as it will break all contracts that use aggregators?